### PR TITLE
Text copy

### PR DESCRIPTION
**Pull request by artemnovichkov**
Add deep copy for text objects.

### CODE REVIEW

File: Texstyle/Texstyle/Text.swift [hunk 1]
```diff
@@ -153,13 +153,24 @@ public final class Text {
         }
         return attributedString.boundingRect(with: size, options: options, context: context)
     }
+
+    public func copy() -> Text {
+        let stylesCopy = styles.mapValues { style in
+            style.copy()
+        }
```
Comment (Antowkos) [inline Texstyle/Texstyle/Text.swift]: Is it possible to do just `let stylesCopy = styles.copy()`?

File: Texstyle/Texstyle/Text.swift [hunk 2]
```diff
@@ -153,13 +153,24 @@ public final class Text {
...
+        let copy = Text(value: value, styles: stylesCopy)
+        copy.substyles = substyles.map { substyle in
+            substyle.copy()
+        }
```
Comment (Antowkos) [inline Texstyle/Texstyle/Text.swift]: Same here for `substyles`?

File: Texstyle/Texstyle/TextStyle.swift [hunk 3]
```diff
@@ -316,7 +316,7 @@ open class TextStyle {
 extension TextStyle: Equatable {
 
     public static func == (lhs: TextStyle, rhs: TextStyle) -> Bool {
-        return lhs.font == rhs.font &&
+        lhs.font == rhs.font &&
             lhs.color == rhs.color &&
             lhs.backgroundColor == rhs.backgroundColor &&
             lhs.ligature == rhs.ligature &&
```
Comment (Antowkos) [inline Texstyle/Texstyle/TextStyle.swift]: Seems like wrong alignment here 🤔

File: Texstyle/Texstyle/Text.swift (hunk 2 above)
Comment (artemnovichkov) [reply Texstyle/Texstyle/Text.swift]: Unfortunately, no:
> Value of type '[TextSubstyle]' has no member 'copy'

File: Texstyle/Texstyle/Text.swift (hunk 1 above)
Comment (artemnovichkov) [reply Texstyle/Texstyle/Text.swift]: No, especially because of `mapValues` function

**Pull request merged**